src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -44,57 +48,182 @@ export class PermissionsTableComponent implements OnInit {
   public entityType: string;
 
   public roles: string[];
-  public entities: Entity[];
+  public entities: Permission[];
   public otherEntities: string[];
 
-  public constructor(private rolesService: RolesService) {
+  public addedRoles: string[] = [];
+  public rolesCheckbox;
+
+  public constructor(private organizationService: OrganizationService,
+    private projectService: ProjectService) {
   }
 
   public ngOnInit() {
+    this.setPermissionsForResource(this.resourceType);
+
     this.roles = ROLES[this.resourceType];
-    this.entities = this.getEntities(this.entityType);
-    this.otherEntities = this.getOtherEntities(this.entityType);
+
+    this.rolesCheckbox = {};
+    this.roles.forEach(r => {
+      this.rolesCheckbox[r] = false;
+    });
   }
 
-  private getEntities(entityType: string): Entity[] {
-    switch (entityType) {
-      case 'users':
-        // TODO get from rolesService instead
-        return [
-          {name: 'marvec', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'livthomas', roles: ['read', 'share', 'write']}
-        ];
-      case 'groups':
-        // TODO get from rolesService instead
-        return [
-          {name: 'managers', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'employees', roles: ['read', 'share']},
-          {name: 'contractors', roles: ['read']}
-        ];
-      default:
-        throw Error('unknown entity type');
+  private getEntities(entityType: string, permissions: Permissions): Permission[] {
+    let entities = permissions[entityType];
+    if (entities) {
+      return entities;
+    } else {
+      throw Error('unknown entity type');
+    }
+  }
+
+  private setPermissionsForResource(resourceType: string) {
+    let parent = this;
+    let getPermissionsFunctions = {
```

Comment:
Does the method `getPermissionsFunctions()` have to be inlined inside `getPermissionsFunctions`?
I see no reason it couldn't be used anywhere else.

Also you can remove the `let parent = this;` and use arrow functions instead of `function() {...}`

Reply:
Not sure what you mean by this comment from the beginning...

And if I'm not mistaken `this` inside an inlined function - as it is in this case - is something different as what I want so I have to keep it in another variable. And when I only use ex. `this.organizationService.getPermissions` there is an error that `HttpClient` object was not created correctly.

Reply:
Well, the whole path with defining a mapper `resourceType` -> `getPermissions` function and then calling that function can be extracted to a method ~`getPermissionsFunctions()`~ `getResourcePermissions()`.

Yes, `this` inside anonymous function doesn't refer to the object. That's why you should change it to arrow functions:
`function() {return parent.organizationService.getPermissions();}` -> `() => this.organizationService.getPermissions()`